packages/fulfillment/integration-tests/__tests__/fulfillment-module-service.spec.ts

```diff
@@ -249,6 +249,200 @@ moduleIntegrationTestRunner({
             )
           })
         })
+
+        describe("shipping options", () => {
+          it("should list shipping options with a filter", async function () {
+            const fulfillmentSet = await service.create({
+              name: "test",
+              type: "test-type",
+              service_zones: [
+                {
+                  name: "test",
+                },
+              ],
+            })
+
+            const shippingProfile = await service.createShippingProfiles({
+              name: "test",
+              type: "default",
+            })
+
+            // TODO: change that for a real provider instead of fake data manual inserted data
+            const [{ id: providerId }] =
+              await MikroOrmWrapper.forkManager().execute(
+                "insert into service_provider (id) values ('sp_jdafwfleiwuonl') returning id"
+              )
+
+            const commonShippingOptionData = {
+              service_zone_id: fulfillmentSet.service_zones[0].id,
+              shipping_profile_id: shippingProfile.id,
+              service_provider_id: providerId,
+              type: {
+                code: "test-type",
+                description: "test-description",
+                label: "test-label",
+              },
+              data: {
+                amount: 1000,
+              },
+            }
+
+            const [shippingOption1] = await service.createShippingOptions([
+              {
+                name: "test-option",
+                price_type: "flat",
+                ...commonShippingOptionData,
+                rules: [
+                  {
+                    attribute: "test-attribute",
+                    operator: "in",
+                    value: ["test-value"],
+                  },
+                ],
+              },
+              {
+                name: "test-option-2",
+                price_type: "flat",
+                ...commonShippingOptionData,
+                rules: [
+                  {
+                    attribute: "test-attribute2",
+                    operator: "eq",
+                    value: "test",
+                  },
+                ],
+              },
+            ])
+
+            const listedOptions = await service.listShippingOptions({
+              name: shippingOption1.name,
+            })
+
+            expect(listedOptions).toEqual(
```

Comment:
You can also just do `expect(listedOptions[0].id).toEqual(shippingOption1.id)`, it's a bit easier to read. Also we can just check `expect(listedOptions).toHaveLength(1)` as we are not doing that check currently.

Reply:
I don't want to just check the length to be sure it is the right one that is returned, but I can definately just check the id ahah

Reply:
Yes sorry if I wasn't clear, but yeah I ll do both, I just didn't want to `just` check the length :D

Reply:
Ah yes, definitely!